### PR TITLE
[kubeadm] improve validation message for join command

### LINKED ISSUE #1513: Join command error message suggests the use of a flag that does not exist
## What keywords did you search in kubeadm issues before filing this one?
kubeadm join command error unsafeSkipCAVerification

## Is this a BUG REPORT or FEATURE REQUEST?

Choose one: BUG REPORT

## Versions

**kubeadm version** (use `kubeadm version`):
```
kubeadm version: &version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-08T17:08:49Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Kubernetes version** (use `kubectl version`): `v1.14.1`
- **Cloud provider or hardware configuration**: AWS EC2
- **OS** (e.g. from /etc/os-release): RHEL 7.6
- **Kernel** (e.g. `uname -a`):

## What happened?
I attempted to join a node without caCertHashes. The join command returned an error that suggested using an unknown flag: `--unsafeSkipCAVerification`

```
ec2-user@ip-172-31-46-205 ~]$ kubeadm join 172.31.46.19:6443 --token m2f82u.k0qdvkv2qtln898s
discovery.bootstrapToken: Invalid value: "": using token-based discovery without caCertHashes can be unsafe. Set unsafeSkipCAVerification to continue
[ec2-user@ip-172-31-46-205 ~]$ kubeadm join 172.31.46.19:6443 --token m2f82u.k0qdvkv2qtln898s --unsafeSkipCAVerification
Error: unknown flag: --unsafeSkipCAVerification
```

## What you expected to happen?
The join command should suggest the correct flag: `--discovery-token-unsafe-skip-ca-verification`

## How to reproduce it (as minimally and precisely as possible)?
Attempt to join a node using a token without providing the CA cert hashes.

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Improve validation message for join command

**Which issue(s) this PR fixes**:

Fixes kubernetes/kubeadm#1513

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/apis/kubeadm/validation/validation.go [hunk 1]
```diff
@@ -146,7 +146,8 @@ func ValidateDiscoveryBootstrapToken(b *kubeadm.BootstrapTokenDiscovery, fldPath
 	}
 
 	if len(b.CACertHashes) == 0 && !b.UnsafeSkipCAVerification {
-		allErrs = append(allErrs, field.Invalid(fldPath, "", "using token-based discovery without caCertHashes can be unsafe. Set unsafeSkipCAVerification to continue"))
+		allErrs = append(allErrs, field.Invalid(fldPath, "",
+			"using token-based discovery without caCertHashes can be unsafe. Set unsafeSkipCAVerification as true in your kubeadm config file or pass --discovery-token-unsafe-skip-ca-verification flag to continue"))
```
Comment (yagonobre) [inline cmd/kubeadm/app/apis/kubeadm/validation/validation.go]: Can you remove this new line?

**Review by yagonobre — COMMENTED** [review]
Thanks, just a nit.
/priority important-longterm
/kind bug

**Review by yagonobre — COMMENTED** [review]
/lgtm

**Review by neolit123 — COMMENTED** [review]
thanks
/approve

**Pull request merged**